DOC: Add FAQ and another JIT batcher illustration to documentation (#2400)
bitburner-official/bitburner-src: src/Documentation/doc/en/help/beginner_faq.md

# FAQs for beginners

This page is dedicated for questions often asked by beginners as they explore the game.

The following are advice, not a bible, as this is a single-player game, so you are your own boss.

#### How do I create directories/folders?

> These are a part of filenames, and therefore nonexistent, files with the same prepending "directory" string are bundled together.

Reviewer: ```suggestion
These are just a part of the filename. There is no explicit command to create them, rather files with the same prepending "directory" string are bundled together.
```

#### Should I backdoor all servers?

Short answer:
Time-consuming for minimal benefits.

Long answer:

> Backdooring a server takes a certain amount of time, and the duration is dependent on a given server's and your hacking level at the time of sending backdoor command.

Reviewer: ```suggestion
Backdooring a server takes a certain amount of time, and the duration is dependent on your hacking level and the given server's at the time of sending backdoor command.
```

Given the amount of servers in the network and how often you have to do so after installing augments, it is ill-advised to perform such potentially lengthy tasks unnecessarily.

#### How do I backdoor through scripts?

> It is only possible in late game, or via exploits.

Reviewer: ```suggestion
It is only possible with an endgame ability, or via exploits.
```
Make it clear that this isn't *automatically* possible in late game.

#### Should I target multiple servers or just one?

> [EHTs (Early Hacking Templates)](../programming/hackingalgorithms.md) can somewhat benefit this at very late stages of the game where the player's hacking level is so high a few threads of `ns.hack()` can take all money of even strongest servers within the network.

Reviewer: ```suggestion
[EHTs (Early Hacking Templates)](../programming/hackingalgorithms.md) can somewhat benefit from this at very late stages of the game where the player's hacking level is so high a few threads of `ns.hack()` can take all money of even strongest servers within the network.
```

However, targetting multiple servers is not viable for batchers in particular due to the fact such algorithms do not suffer from over-hacking (a loose term that describes a targeted server being drained of all reserved money).

#### What are the differences between threads and processes?

> Threads are a part of a process, processes have their own thread count.

Reviewer: Although I don't have a good suggestion, I think this answer is still confusing.

Reviewer 2: This is my suggestion combined with fico's one:
```
When you run a script, you spawn a "process" that runs your code. With each process, you can specify a number of threads (default to 1). "Thread" is simply a multiplier applied to the effect of some actions (e.g., hack, grow, weaken). Using more threads multiplies the effect, but it also multiplies the RAM cost of running the script.
```
